test/Microsoft.ML.Core.Tests/UnitTests/TestModelLoad.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.TestFramework;
+using System.IO;
+using Xunit;
+
+namespace Microsoft.ML.Runtime.RunTests
+{
+    public class TestModelLoad
+    {
+        /// <summary>
+        /// Tests loading a model file that was saved using an older version still loads correctly.
+        /// </summary>
+        [Fact]
+        public void LoadOriginalBinaryLoaderModel()
+        {
+            using (var env = new LocalEnvironment()
+                .AddStandardComponents())
+            using (var modelStream = File.OpenRead("OriginalVersionBinaryLoader.zip"))
```

Comment:
I might recommend naming this model w/ the version number which created it. Saying 'OriginalVersion' makes we think we'll have a plethora of non-comparable names like 'NewVersion', and 'NewVersionAfterUpdate'. 

```suggestion
            using (var modelStream = File.OpenRead("BinaryLoader-v3.11.0.0.zip"))
```

Reply:
Good idea. I need to update the TestModels NuGet package, which I'm doing here: https://github.com/dotnet/machinelearning-testdata/pull/8/.

Once that is approved, merged, and published, I will update this PR.